internal/injector/aspect/advice/code/dot_ast.go

```diff
@@ -0,0 +1,12 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+//go:generate go run ./generator -o dot_ast.proxies.go
```

Comment:
Generates proxies that essentially ensure we have the correct `.String() string` implementation to "move" existing nodes into the generated code. This allows the templates to traverse the AST transparently.